tinkerpop3/src/test/java/io/shiftleft/overflowdb/IndexesTest.java

```diff
@@ -75,5 +77,36 @@ public void shouldUseIndicesCreatedBeforeLoadingData() throws IOException {
         avgTimeWithIndex < avgTimeWithoutIndex);
   }
 
+  @Test
+//  @Ignore // only run manually since the timings vary depending on the environment
+  public void shouldUseStoredIndices() throws IOException {
+    int loops = 100_000;
+    final double avgTimeWithIndexCreation;
+    final double avgTimeWithStoredIndex;
+    final File overflowDb = Files.createTempFile("overflowdb", "bin").toFile();
+    overflowDb.deleteOnExit();
+
+    { // tests with index
+      OdbGraph graph = GratefulDead.newGraphWithData(overflowDb.getAbsolutePath());
+      graph.indexManager.createNodePropertyIndex("performances");
+      GraphTraversalSource g = graph.traversal();
+      assertEquals(142, (long) g.V().has("performances", P.eq(1)).count().next());
+      avgTimeWithIndexCreation = TimeUtil.clock(loops, () -> g.V().has("performances", P.eq(1)).count().next());
+      graph.close();
+    }
+
+    { // tests with stored index
+      OdbGraph graph = GratefulDead.newGraph(OdbConfig.withDefaults().withStorageLocation(overflowDb.getAbsolutePath()));
+      GraphTraversalSource g = graph.traversal();
+      assertEquals(142, (long) g.V().has("performances", P.eq(1)).count().next());
+      avgTimeWithStoredIndex = TimeUtil.clock(loops, () -> g.V().has("performances", P.eq(1)).count().next());
+      graph.close();
+    }
+
+    System.out.println("avgTimeWithIndexCreation = " + avgTimeWithIndexCreation);
+    System.out.println("avgTimeWithStoredIndex = " + avgTimeWithStoredIndex);
+    assertTrue("avg time with index should be (significantly) less than without index",
+        avgTimeWithIndexCreation > avgTimeWithStoredIndex);
+  }
```

Comment:
Please also add a test that asserts the presence of an index in an way that not dependent on the environment (but keep this one, to ensure we can also test it end to end). 

I mean something like 
* setup odb with index
* close, restart
* graph.indexManager.getIndex("abc").count shouldBe 50

That one we don't need to leave as 'ignored' and it'll be easier to debug, in case it goes red.

Reply:
Done